gr-digital/lib/costas_loop_cc_impl.cc

```diff
@@ -175,6 +179,15 @@ int costas_loop_cc_impl::work(int noutput_items,
                       nitems_read(0) + noutput_items,
                       pmt::intern("phase_est"));
 
+    // Get this out of the for loop if not used:
+    bool hasAdditionalOutputs = false;
```

Comment:
if the number of streams don't change during runtime, then this value can be computed elsewhere & doesn't need to be done every single call to work. I think we (ab)use `::check_topology` for this purpose.

Reply:
I'm also not convinced of the benefit of this variable, later in the work function. If none of the streams are in use then it's 1 `if` & otherwise 4. Since we know the number of streams before any of this happens, we should be able to make optimal use of that knowledge here. Worth considering IMHO.

Reply:
@michaelld I did look at the API doc and the source code for check_topology and it isn't clear on the meaning of valid combination.  It would be great to check the actual connected outputs in the constructor and set it once then use it in work.  But I couldn't see an easy way to do that?

Reply:
basically `check_topology` will be provided with the hoped-for # of input and output connections & returns whether that topology meets the blocks needs. For this block, it has better be 1 input & 1-3 outputs. So long as those provided meet the criteria of the block, return true; otherwise return false. AND, since the topology won't change once set, in that method you can setup whatever else you want to handle the possibly variable output topology. I'm pretty sure some of the audio sources / sinks do this. `grep` the source for `check_topology` to get some examples ...